### PR TITLE
Add ElasticSearchClient and binding

### PR DESCRIPTION
**Pull request by duketon**

### CODE REVIEW

File: src/main/scala/com/evojam/play/elastic4s/client/ElasticSearchClientImpl.scala [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    }
+
+  override def search(searchDef: SearchDefinition)(implicit exc: ExecutionContext) =
+    elastic.execute(searchDef)
+
+  override def indexDocument[T](indexName: String, doctype: String, id: String, doc: T)
+    (implicit w: Writes[T], exc: ExecutionContext) = {
+    require(indexName != null, "indexName cannot be null")
+    require(doctype != null, "doctype cannot be null")
+    require(doc != null, "doc cannot be null")
```
Comment (albsadowski) [inline src/main/scala/com/evojam/play/elastic4s/client/ElasticSearchClientImpl.scala]: why you are checking not null inside of the method?

it that's necessary then write a comment why.

File: src/main/scala/com/evojam/play/elastic4s/client/ElasticSearchClientImpl.scala [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    (implicit exc: ExecutionContext) =
+    elastic.execute {
+      logger.debug(s"Update document index=$indexName type=$doctype id=$id doc=${in.json}")
+      update
+        .id(id)
+        .in(indexName -> doctype)
+        .doc(in)
+    }
+
+  override def updateDocument[T](indexName: String, doctype: String, id: String, in: T)(implicit w: Writes[T], exc: ExecutionContext) = { //scalastyle:ignore
```
Comment (albsadowski) [inline src/main/scala/com/evojam/play/elastic4s/client/ElasticSearchClientImpl.scala]: break line instead of scalastyle:ignore

**Comment by albsadowski** [comment]
LGBT

**Comment by abankowski** [comment]
LGTM

**Pull request merged**